mmcls/evaluation/metrics/multi_task.py

```diff
@@ -30,53 +84,64 @@ def process(self, data_batch, data_samples: Sequence[dict]):
             data_batch: A batch of data from the dataloader.
             data_samples (Sequence[dict]): A batch of outputs from the model.
         """
+        data_sample_instances = []
+        for data_sample in data_samples:
+            if 'gt_task' in data_sample:
+                data_sample_instances.append(MultiTaskDataSample().set_gt_task(
+                    data_sample['gt_task']).set_pred_task(
+                        data_sample['pred_task']))
         for task_name in self.task_metrics.keys():
-            index = list(self.task_metrics.keys()).index(task_name)
-            task_data_samples = []
-            for data_sample in data_samples.copy():
-                if task_name in data_sample['gt_label'].keys():
-                    data_sample['gt_label'] = {
-                        'label': data_sample['gt_label'][task_name]
-                    }
-                    if len(data_sample['pred_label']['score']) < 3:
-                        raise Exception('score masked')
-                    data_sample['pred_label']['score'] = data_sample[
-                        'pred_label']['score'][index]
-                    task_data_samples.append(data_sample)
-            globals()['metric_%s' % task_name].process(data_batch,
-                                                       task_data_samples)
+            filtered_data_samples = []
+            for data_sample in data_sample_instances:
+                sample_mask = data_sample.get_task_mask(task_name)
+                if sample_mask:
+                    filtered_data_samples.append(data_sample)
+            if type(self.task_metrics[task_name]) != dict:
+                for metric in self._metrics[task_name]:
+                    if metric.__class__.__name__ != 'MultiTasksMetric':
```

Comment:
I think we should add a comment here
```suggestion
                    # Current implementation is only comptaible 
                    # With 2 types of metrics : 
                    # * Cls Metrics 
                    # * Nested Cls Metrics
                    # In order to make it work with other
                    # non-cls heads/metrics, one will have to 
                    # override the current implementation
                    if metric.__class__.__name__ != 'MultiTasksMetric':
```